[wip][quantization] Enable OSS quantization tests

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23718 [wip][quantization] Enable OSS quantization tests**
* #23837 Add flag to temporarily disable MKL-DNN conv

Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [x] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
[quantization] Enable OSS quantization tests

raghuramank100 commented on test/run_test.py:
```diff
@@ -38,9 +39,14 @@
     'multiprocessing_spawn',
     'nccl',
     'nn',
+    'nn_quantized',
```
Please add qat to this list.

raghuramank100 reviewed (approved):
Please add one more test (see comments), thanks for fixing this issue.

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"

[quantization] Enable OSS quantization tests

ghstack-source-id: 4a7303a62f55fafd5b9fb68be3ee76cdecd37e51
Pull Request resolved: https://github.com/pytorch/pytorch/pull/23718

gh-metadata: pytorch pytorch 23718 gh/jamesr66a/34/head

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [ ] Range of scale needs to be adjusted in `hu.qparams`

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [ ] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [ ] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "[wip][quantization] Enable OSS quantization tests"


Changes:

- Enable tests for quantization test files in `run_tests.py`
- Remove `__future__` imports from `torch/nn/qat/modules/__init__.py`, since `unicode_literals` messes up imports on python2 because the elements in `__all__` will be Unicode and not string
- Skip PostTrainingQuantTests if the build doesn't have FBGEMM (only a small subset of targets in tests) or if testing under UBSAN (the suppression file doesn't seem to work)
- Initialize the CUDA runtime in a static class method on `TestFakeQuantizePerTensorAffine`. This prevents CUDA test cases from timing out because the runtime initialization time is accounted for in a test

Outstanding issues:

- [x] MKLDNN corruption, see https://github.com/pytorch/pytorch/issues/23763#issuecomment-517958683
- [x] There is an issue with the CUDA fake quantization tests causing flaky failures

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)
